src/config.rs

```diff
@@ -495,17 +486,30 @@ impl Cfg {
         Ok(None)
     }
 
-    pub fn find_override_toolchain_or_default(
+    pub fn find_or_install_override_toolchain_or_default(
         &self,
         path: &Path,
     ) -> Result<Option<(Toolchain<'_>, Option<OverrideReason>)>> {
-        Ok(
+        if let Some((toolchain, reason)) =
```

Comment:
This feels a smidge awkward as a double-nested `if let` -- could it be split out a bit?  I know there's a tendency to prefer expressions over statements in Rust code, but this is just a little too nested for my brain to grok easily.